src/Development/IDE/Core/Rules.hs

```diff
@@ -667,10 +702,14 @@ getModIfaceRule = define $ \GetModIface f -> do
             case mb_pm of
                 Nothing -> return (diags, Nothing)
                 Just pm -> do
-                    (diags', tmr) <- typeCheckRuleDefinition f pm DoGenerateInterfaceFiles
-                    -- Bang pattern is important to avoid leaking 'tmr'
-                    let !res = extract tmr
-                    return (diags <> diags', res)
+                    (diags, mb_hsc) <- ghcSessionDepsDefinition f
```

Comment:
Why do we not use the cached version here?

Reply:
To avoid creating expensive cache entries for non-files of interest. Will add a comment